### PR TITLE
[Improve][Connector-V2][StarRocks] Unified exception for StarRocks so…

### PR DESCRIPTION
**Pull request by john8628**
…urce & sink connector

## Purpose of this pull request

https://github.com/apache/incubator-seatunnel/issues/3545
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: docs/en/connector-v2/Error-Quick-Reference-Manual.md [hunk 1]
```diff
@@ -89,6 +89,7 @@ problems encountered by users.
 |----------------|---------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------|
 | TABLESTORE-01  | Failed to send these rows of data     | When users encounter this error code, it means that failed to write these rows of data, please check the rows that failed to import |
 
+
```
Comment (TaoZex) [inline docs/en/connector-v2/Error-Quick-Reference-Manual.md]: Remove

File: docs/en/connector-v2/Error-Quick-Reference-Manual.md [hunk 2]
```diff
@@ -144,6 +145,7 @@ problems encountered by users.
 | IOTDB-02 | Initialize IoTDB client failed                           | When the user encounters this error code, it indicates that the client initialization failed. Please check   |
 | IOTDB-03 | Close IoTDB client failed                                | When the user encounters this error code, it indicates that closing the client failed. Please check          |
 
+
```
Comment (TaoZex) [inline docs/en/connector-v2/Error-Quick-Reference-Manual.md]: Remove

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksStreamLoadVisitor.java [hunk 3]
```diff
@@ -165,12 +167,12 @@ private void checkLabelState(String host, String label) throws IOException {
                 String queryLoadStateUrl = new StringBuilder(host).append("/api/").append(sinkConfig.getDatabase()).append("/get_load_state?label=").append(label).toString();
                 Map<String, Object> result = httpHelper.doHttpGet(queryLoadStateUrl, getLoadStateHttpHeader(label));
                 if (result == null) {
-                    throw new IOException(String.format("Failed to flush data to StarRocks, Error " +
+                    throw new StarRocksConnectorException(CommonErrorCode.ILLEGAL_ARGUMENT, String.format("Failed to flush data to StarRocks, Error " +
```
Comment (TaoZex) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksStreamLoadVisitor.java]: Why use CommonErrorCode.ILLEGAL_ARGUMENT？

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksStreamLoadVisitor.java [hunk 4]
```diff
@@ -165,12 +167,12 @@ private void checkLabelState(String host, String label) throws IOException {
...
                             "could not get the final state of label[%s].\n", label), null);
                 }
                 String labelState = (String) result.get("state");
                 if (null == labelState) {
-                    throw new IOException(String.format("Failed to flush data to StarRocks, Error " +
+                    throw new StarRocksConnectorException(CommonErrorCode.ILLEGAL_ARGUMENT, String.format("Failed to flush data to StarRocks, Error " +
```
Comment (TaoZex) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksStreamLoadVisitor.java]: Same as above.

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/sink/StarRocksSinkWriter.java [hunk 5]
```diff
@@ -73,7 +75,7 @@ public void close() throws IOException {
             }
         } catch (IOException e) {
             log.error("Close starRocks manager failed.", e);
-            throw new IOException("Close starRocks manager failed.", e);
+            throw new StarRocksConnectorException(CommonErrorCode.SQL_OPERATION_FAILED, e);
```
Comment (TaoZex) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/sink/StarRocksSinkWriter.java]: Close manage failures have nothing to do with SQL, and it is better to define the associated error code.

File: docs/en/connector-v2/Error-Quick-Reference-Manual.md (hunk 1 above)
Comment (john8628) [reply docs/en/connector-v2/Error-Quick-Reference-Manual.md]: ok, thank you for your comment;

File: docs/en/connector-v2/Error-Quick-Reference-Manual.md (hunk 2 above)
Comment (john8628) [reply docs/en/connector-v2/Error-Quick-Reference-Manual.md]: ok, thank you for your comment;

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksStreamLoadVisitor.java (hunk 3 above)
Comment (john8628) [reply seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksStreamLoadVisitor.java]: I am very confused about the same ILLEGAL_ARGUMENT having diff places, with the same meaning;

Comment (TaoZex) [reply seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksStreamLoadVisitor.java]: You can use the prompts to know which error code to use.In this case, the error code should be related to `flush data fail`.

**Comment by TaoZex** [comment]
https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/setup.md#install-jetbrains-idea-checkstyle-idea-plugin

Please change your code style through this document.

**Comment by john8628** [comment]
> 



> https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/setup.md#install-jetbrains-idea-checkstyle-idea-plugin
> 
> Please change your code style through this document.

yean , I will merge lastest dev , and check the code style ,thanks

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/exception/StarRocksConnectorErrorCode.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.starrocks.exception;
+
+import org.apache.seatunnel.common.exception.SeaTunnelErrorCode;
+
+public enum StarRocksConnectorErrorCode implements SeaTunnelErrorCode {
+  FLUSH_DATA_FAILED("STARROCKS-01", "Flush batch data to sink connector failed"),
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/exception/StarRocksConnectorErrorCode.java:23]: Remove this useless error code. Use `CommonErrorCode.FLUST_DATA_FAILED`

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/exception/StarRocksConnectorErrorCode.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    private final String description;
+
+    StarRocksConnectorErrorCode(String code, String description) {
+        this.code = code;
+        this.description = description;
+    }
+
+    @Override
+    public String getCode() {
+        return null;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/exception/StarRocksConnectorErrorCode.java]: return code

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/exception/StarRocksConnectorErrorCode.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Override
+    public String getDescription() {
+        return null;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/exception/StarRocksConnectorErrorCode.java]: return decription

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksSinkManager.java [hunk 9]
```diff
@@ -116,10 +118,10 @@ public synchronized void flush() throws IOException {
             } catch (Exception e) {
                 log.warn("Writing records to StarRocks failed, retry times = {}", i, e);
                 if (i >= sinkConfig.getMaxRetries()) {
-                    throw new IOException("Writing records to StarRocks failed.", e);
+                    throw new StarRocksConnectorException(StarRocksConnectorErrorCode.WRITE_RECORDS_FAILED, e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksSinkManager.java]: throw new StarRocksConnectorException(StarRocksConnectorErrorCode.WRITE_RECORDS_FAILED,  errorMessage, e);

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksSinkManager.java [hunk 10]
```diff
@@ -143,7 +144,7 @@ public synchronized void flush() throws IOException {
 
     private void checkFlushException() {
         if (flushException != null) {
-            throw new RuntimeException("Writing records to StarRocks failed.", flushException);
+            throw new StarRocksConnectorException(StarRocksConnectorErrorCode.FLUSH_DATA_FAILED, flushException);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksSinkManager.java:147]: throw new StarRocksConnectorException(StarRocksConnectorErrorCode.FLUSH_DATA_FAILED, errorMessage, flushException)

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/sink/StarRocksSinkWriter.java [hunk 11]
```diff
@@ -84,6 +86,6 @@ public static StarRocksISerializer createSerializer(SinkConfig sinkConfig, SeaTu
         if (SinkConfig.StreamLoadFormat.JSON.equals(sinkConfig.getLoadFormat())) {
             return new StarRocksJsonSerializer(seaTunnelRowType);
         }
-        throw new RuntimeException("Failed to create row serializer, unsupported `format` from stream load properties.");
+        throw new StarRocksConnectorException(CommonErrorCode.SERIALIZE_OPERATION_FAILED, "Failed to create row serializer, unsupported `format` from stream load properties.");
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/sink/StarRocksSinkWriter.java]: CommonErrorCode.ILLEGAL_ARGUMENT is better

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksSinkManager.java [hunk 12]
```diff
@@ -116,10 +119,10 @@ public synchronized void flush() throws IOException {
             } catch (Exception e) {
                 log.warn("Writing records to StarRocks failed, retry times = {}", i, e);
                 if (i >= sinkConfig.getMaxRetries()) {
-                    throw new IOException("Writing records to StarRocks failed.", e);
+                    throw new StarRocksConnectorException(StarRocksConnectorErrorCode.WRITE_RECORDS_FAILED, "Writing records to StarRocks failed.", e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/client/StarRocksSinkManager.java]: throw new StarRocksConnectorException(StarRocksConnectorErrorCode.WRITE_RECORDS_FAILED, "The number of retries was exceeded, writing records to StarRocks failed.", e)

**Review by TyrantLucifer — DISMISSED** [review]
LGTM

**Review by PraveenNanda124 — COMMENTED** [review]
Good work

**Review by TaoZex — DISMISSED** [review]
LGTM
Please wait for https://github.com/apache/incubator-seatunnel/pull/3630 merged and rerun CI.

**Review by TaoZex — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**